### ISSUE TITLE
Netlify adapter config sources

### ISSUE THREAD

**Issue by rsdavis** [sveltejs/kit#1604]
**Is your feature request related to a problem? Please describe.**
Yes. In a monorepo setup, it is common to put the frontend client code in its own directory. But, if you want to configure the base directory for Netlify, the `netlify.toml` file must go in the root directory. The problem is that the SvelteKit Netlify adapter looks for the config file in the base directory. Of course, its is not there, and an error is thrown "Missing a netlify.toml file."

Example directory structure where `site/` is the base directory ...
```
backend/
site/
site/package.json
netlify.toml
```
```
# netlify.toml
[build]
  base      = "site/"
  command   = "npm run build"
  publish   = "build/"
  functions = "functions/"
```

**Describe the solution you'd like**
It would be nice to be able to feed these parameters into the adapter. So the config file would look something like ...
```
const config = {
    kit: {
        target: '#svelte',
        adapter: netlify({
           publish: "build",
           functions: "functions" 
       })
    }
};
```
I also imagine that there would be default values in the case when the parameters are not specified and the `netlify.toml` is not present. With the approach, the `netlify.toml` file can be kept at root level, and the adapter still gets the information that it needs without the file needing to present in the base directory.

**How important is this feature to you?**
This feature is not critical because you can get around this issue by setting the base directory in the Netlify UI and create a `netlify.toml` file in the base directory that supplies the parameters. But its is important because setting parameters via the UI is not versioned or easily reproducible. Committing the base path via config is essential for achieving infrastructure-as-code.

**Additional context**
I searched this issue and could not find it being discussed anywhere. My apologies if I am missing something. I am loving SvelteKit so far, excellent work, thanks!

**Comment by jthegedus** [comment]
Most platforms Adapters target have their own config files which we want to read to calculate output dirs for SvelteKit builds. Instead of solving this issue ad-hoc, passing different sets of params to the different adapters, I would suggest we pass the config file location, then use that as the root for the computed dest dirs. I have had success with this in the [Firebase Adapter](https://github.com/jthegedus/svelte-adapter-firebase). 

Thus my suggestion would be:

```
backend/
site/
  package.json
  svelte.config.js
netlify.toml
```

```js
// svelte.config.js
const config = {
    kit: {
        target: '#svelte',
        adapter: netlify({
           netlifyConfig: "../netlify.toml"
       })
    }
};
```

```toml
# netlify.toml
[build]
  base      = "site/"
  command   = "npm run build"
  publish   = "build/"
  functions = "functions/"
```

with the output looking like this:

```
build/        <-- computed output dirs relative from netlify.toml
functions/    <-- computed output dirs relative from netlify.toml
backend/
site/
  .svelte-kit
  package.json
  svelte.config.js
netlify.toml
```

**Comment by sw-yx** [comment]
i think @jthegedus' solution makes the most sense. no point inlining the config when you'd have to repeat it again. also recommend calling it `netlifyTomlPath` or `configPath` rather than `netlifyConfig`

**Comment by serhalp** [comment]
@teemingc My reaction to this is that Netlify framework adapters shouldn't need to ever parse `netlify.toml`, I think? So perhaps the focus here should be on seeing that as the problem here and refactoring it away.

It looks like it's being used to get the Netlify publish directory, in order to copy files like `_headers` and `_redirects` into it. This seems backwards to me: a user would set their Netlify project's publish directory _to_ their SvelteKit app's `outDir`. So we should be able to simply write to `outDir`...? 🤔 This is how we do things in other frameworks' Netlify adapters.

**Comment by teemingc** [comment]
Yeah, that sounds good. This would be a breaking change so I've added that tag.

As a side note, we allow configuring the path of the final build output in the Node, static, and Cloudflare adapters. The Vercel adapter doesn't have this option and always outputs to `.vercel`. In Cloudflare's case, there's a user option to specify the path of the `wrangler.jsonc` config file, but we also use the Cloudflare provided utility to find the config file in the current directory and directories above it.
